### DISCUSSION TITLE
Enable or disable default actions per model

### DISCUSSION THREAD

**Discussion by brunoprietog** [avo-hq/avo#645] [Feature requests]
Hi! great gem!

I would like to know if it is possible to disable some actions per model for all users without using pundit. I don't really need advanced authorization, just be able to disable some actions for read-only models for all users.

I also wonder if in the future, in case I implement policies, they would not collide with the ones for Avo? Anyway, I intend to use Action Policy instead of Pundit, as it has an extension to use it with GraphQL, and so I can dispense with GraphQL pro which for me as a student is highly prohitive for the price.

Thanks!

**Comment by adrianthedev** [comment]
Hey Bruno,

Unfortunately, there isn't another way to hide those buttons. Pundit policies work very nice for this scenario. 
We're also not looking to add other methods of hiding buttons because of the maintenance burden. We don't want to maintain two mechanisms that do the same thing. Too much complexity.

When you decide to add policies, they will not really collide with Avo's functionality. There are a few common methods that you can [easily override](https://docs.avohq.io/1.0/authorization.html#using-different-policy-methods), so they don't clash with your own.

I haven't worked with the Action policy package before. However, it looks like it's an excellent package.

Regarding the price, we do have a discount for start-ups, open-source or educational projects, so reach out to us if you need it. The price is currently discounted until we launch all the in-development features.
This package is quite complex, and maintenance takes a lot of our time. I work full-time on it, and it's still far from paying the bills. 
Given how many problems it solves and how quickly it allows developers to build great interfaces, I believe $100 is relatively low. 
I know that not everyone can afford to spend big-corp money, so that's why we have the discount.

Thanks for reaching out!

**Comment by brunoprietog** [comment]
Thanks for your response!

Yes, I think the price is right. I was referring to GraphQL's price which is prohibitive for me at least, USD $900 a year.

I looked into that feature to replace the default policies. The problem is that if I use Action policy, I could not comfortably implement Pundit policies as well because the classes would collide, since both libraries use the same conventions.

That's why I was asking if there is a pundit-free workaround to do that, so that any method that returns true/false can be used without tying to the use of a specific authorization library. It could even be calling a simple method of a model. Would something like that add too much complexity?

**Reply by adrianthedev** [reply]
Oh, ok, got it regarding the price.

I see what you mean about Pundit policies clashing with Action policy. I now wonder if a framework agnostic policy can be implemented. Something that will work with Pundit and Action policy.
TBH it's not something that we have on our roadmap right now, but would welcome if it were available and other users would ask for.
